### ISSUE TITLE
User defined defaults_options are overridden by default defaults_options

### ISSUE THREAD

**Issue by kcem** [gael-ian/vagrant-bindfs#93]
User set default_options value in Vagrantfile are overridden at some point during `vagrant up` and set to default.

Example way to reproduce:
Vagrantfile:
```
Vagrant.configure('2') do |instance|

  instance.vm.box = 'centos/7'

  # provider lookup order
  instance.vm.provider :libvirt
  instance.vm.provider :virtualbox

  instance.bindfs.debug = true
  instance.bindfs.default_options = {
      perms: ''
  }
  instance.bindfs.bind_folder "/vagrant", "/home/vagrant/test"

  instance.vm.provider :libvirt do |v, o|  #<- remove this block and default_options will not be overriden
    v.cpus = 4
  end
end
```

Output:
```
(...)
    default: Bindfs 1.14.2 is installed
==> default: Machine is ready to use bindfs!
==> default: Creating bind mounts after synced_folders...
    default: /vagrant => /home/vagrant/test
    default: /usr/local/bin/bindfs --force-user=vagrant --force-group=vagrant --perms=u=rwX:g=rD:o=rD /vagrant /home/vagrant/test
```

Should be:
```
    (...)
    default: /usr/local/bin/bindfs --force-user=vagrant --force-group=vagrant --perms= /vagrant /home/vagrant/test
```

**Comment by gael-ian** [comment]
Hello @kcem 

Thanks for this reporting and a concise way to reproduce, it helps a lot. I'll try to fix this as soon as I can, potentially this week-end.

**Comment by gael-ian** [comment]
Hello @kcem 

Configuration in Vagrant can be defined in the project Vagrantfile, in a `~/Vagrantfile`, inside a box, and for each machine in every of these files. In total, in a test setup with a single VM and a single shared folder, no less than 19 configuration objects are created and merged.

Accross all those merge, `default_options` set in the top level scope of a project Vagrantfile (`Vagrant.configure('2') do {}`) were quickly overridden by default values from the other locations. I had to switch to another strategy and use the same `UNSET_VALUE`+`finalize!` trick as for `bindfs_version` to be able to keep configured `default_options` from one scope to another.

That solution is more interesting than a class variable for very complex setups, as you can define different sets of options for every scope and have them merged from top to bottom.

The fix is available in `vagrant-bindfs 1.1.7`.

**Comment by kcem** [comment]
Hello @gael-ian 

Thank you for explaination and for fix.
